tests/trainer/test_signal_handling.py

```diff
@@ -0,0 +1,66 @@
+import os
+import platform
+import signal
+
+import pytest
+import torch.distributed
+
+from pytorch_lightning import Trainer, Callback
+from pytorch_lightning.trainer.states import TrainerState
+from tests.base import EvalModelTemplate
+
+
+class KillCallback(Callback):
+    """ A callback that simulates a terminal signal (e.g. SIGINT). """
+
+    def __init__(self, signal_code):
+        self._signal = signal_code
+
+    def on_batch_end(self, trainer, pl_module):
+        # send the signal after the first batch
+        assert trainer.global_step == 0, "did not interrupt training after first batch"
+        pid = os.getpid()
+        os.kill(pid, self._signal)
+
+    def on_train_end(self, trainer, pl_module):
+        assert trainer.train_loop._teardown_already_run
+
+    def on_keyboard_interrupt(self, trainer, pl_module):
+        assert trainer.state == TrainerState.INTERRUPTED
+        assert trainer.interrupted
+
+
+def get_available_signal_codes():
```

Comment:
```suggestion
def _get_available_signal_codes():
```
let shave as protected, to direntiate from standard public PL api

Reply:
> also, this shall favor multiple training with a single Trainer in one scriop/execution, right?

yes, this is fine.